### DISCUSSION TITLE
What is rocket / Landing Page / Marketing

### DISCUSSION THREAD

**Discussion by daKmoR** [modernweb-dev/rocket#282] [General]
# What rocket is about

Slogan direction ideas
- Rocket enables everyone to code a website
- Rocket enables all disciplines like Programmer, Designer, Marketer, ... to work together
- It does not feel like programming
- Fast and small websites for everyone (even on old phones or with a crappy internet connection)

## Target group

Initially, it will probably be more steered towards programmers as it will require some knowledge to set up. Which we can later reduce to bring in more disciplines.

In the end (2023+), it should be approachable for the secretary, the marketer, the programmer, ... to work together in a "single" place.

## Landing Page

See current WIP landing page: https://next.rocket.modern-web.dev

these are very rough ideas
1. You own the content (as files) which means that you can host it or move it wherever you want
2. You will have access to all the versions ever "saved" (via git) and you can bring it back individual parts
3. You can invite programmers to add logic into specific files/pages which means that you can display complex info/data (even from external sources)
4. It essentially helps you to "code" without needing to "code" which means that adding complex logic, later on, is straightforward
5. We don't have a custom environment but you work in an editor which means that less maintenance/complexity by use and more power feature for you
6. Enables a rich theme/plugin/preset environment which means that you can pick and choose to combine functionalities
7. The default "theme/preset" (called "Rocket Launch") outputs a very fast and small website which means that it will feel responsive even on older phones or with a crappy internet connection 

**More for programmers**
1. Every content (be it markdown/html or js) does have access to javascript
2. Rockets rendering logic is rather "simple" which means that you can roll your own if needed
3. Rockets rendering It has a low dependency count which means you don't need to keep updating all the time
4. The "biggest" complexity we manage is analyzing your source code to have a smart/fast development server
5. The only "template" system we have is javascript built-in template literals which means that we have fewer dependencies
6. It's a modern es module only codebase
7. You can generate all kinds of output (using frontend frameworks or not)
8. Rocket renders with "less" magic which means that devs get access to all info whichever page file they take as an entrypoint
9. It is sort of a Static Site Generator but it focuses on single page renderings which means that during development only individual pages get rerendered which means that it's fast and responsive

We probably will need to explain some of these details in a visual way to make sure everyone gets the point.
A good example for such explanations would be https://remix.run/

## Market

There are a LOT of different Static Site Generators (SSG) on the market. 

1. [Eleventy](https://www.11ty.dev/) 
   The base of rocket as it is currently released. This part get's replaced in the "new version" by the single page rendering system. None the less it's a huge inspiration and the way it's open and honest should be a role model. The documentation is good but marketing-wise or landing page wise probably not a good example. This is a "low level" SSG e.g. you need to build everything yourself for the output (e.g. framework junk free)
   
2. [NextJs](https://nextjs.org/)
   Not really an SSG but a framework around React. 
   Probably the project with the biggest budget (Powered by a multi-million company).

3. [Gatsby](https://www.gatsbyjs.com/)
   Sort of like Next but for SSG.
   It outputs a React (probably still most hyped js framework) page (which are usually quite big on the javascript side).
   The landing page is quite nice
      
4. [NuxtJs](https://nuxtjs.org/)
   An SSG specific to Vue (another popular js framework)
   
5. [Astro](https://astro.build/)
   A "relatively" new SSG that supports multiple frameworks in the same site (e.g. combine React, Vue, Svelte, ...)
   We could get some selling points and some ideas from there
   Overall the marketing is not that great - but it's still quite popular with 8k+ Github Stars
   
6. [GitBook](https://www.gitbook.com/)
   Probably the product which has the most similar base features. e.g. files => nice docs site
   So probably the most selling points would work for rocket as well (but we have more on top 😬)
   
7. [WordPress](https://wordpress.org/)
   A LOT of WordPress sites should probably be SSGs... so naturally, Rocket could "replicate" a lot of what WordPress is doing.
   WordPress currently powers about ~35% of ALL active internet websites. This is just crazy... so if you could convince some of those users to consider switching or using rocket for their next project that would be huge... 
      
8. [Readme](https://readme.com/)
   Seems very similar GitBook... and they ask for a lot of money to host it... maybe that could be a business model?
   Quite a nice landing page
   
List of a LOT of ssgs https://jamstack.org/generators/

**Comment by petecarapetyan** [comment]
If Rocket could be condensed to a sound bite, what would that sound bit be?

For example, but not proposed as 'the answer' to the above question:

```
ROCKET:
  - smallest fastest dumbest site generator for the widest audience
  - created by removing legacy stacks, adding latest fastest simplest stacks
  ```

**Reply by petecarapetyan** [reply]
What I am shooting for in the above is the magical confidence quotient. 

If I could be confident that an SSG could be counted on at the abstract level to ALWAYS remove legacy junk and ALWAYS replace that with simpler dumber better working stacks, then I wouldn't have to read through detailed descriptions of which stacks are included, which are not, and then one at a time compare each one with the question `hmmm should this stack be included or is yada stack better in it's place right now?` which could mean anything from jsx to esbuild to whatever is on caniuse to ... whatever.

Just like this comment itself, the idea isn't to make a long explanation of each thought, but instead follow the abstract guideline with some degree of integrity.

Example: I had to do a bit of research to figure out that remix was a react consumer. Yeah, eff that.

**Comment by petecarapetyan** [comment]
Also for each of the SSG's+ listed above, 1-7 Next.js Gatsby etc ... I would like to know what the value add is for Rocket over that particular SSG or ... 

By not being explicit here, we risk being 'just another' without any particular reason for reinventing this wheel. May not be obvious to us, but it's the first thing that any passer-by thinks.

**Comment by daKmoR** [comment]
Frameworks have compounding cost
https://glazkov.com/2022/02/23/the-cost-of-opinion/

**Comment by daKmoR** [comment]
the new landing page of @rocket/js seems pretty good - if there are any more improvements needed feel free to open a new issue/discussion.
